File: 프로그래머스/0/181857. 배열의 길이를 2의 거듭제곱으로 만들기/배열의 길이를 2의 거듭제곱으로 만들기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> arr) {
    int n = 0;
    int idx = arr.size();
    while(idx != 1 << n){
        if(idx > 1 << n) n++;
        else if (idx < 1 << n) idx++;
    }
    arr.resize(idx, 0);
    return arr;
}

File: 프로그래머스/0/181857. 배열의 길이를 2의 거듭제곱으로 만들기/배열의 길이를 2의 거듭제곱으로 만들기.java
class Solution {
    public int[] solution(int[] arr) {
        int n = 0;
        int idx = arr.length;
        while(idx != 1 << n){
            if(idx > 1 << n) n++;
            else if (idx < 1 << n) idx++;
        }
        int[] answer = new int[idx];
        for (int i = 0; i < arr.length; i++) {
            answer[i] = arr[i];
        }
        return answer;
    }
}